src/System.Data.OleDb/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -0,0 +1,1531 @@
+                // FileIOPermission.RevertAssert();
+            }
+        }
+
+        [ResourceExposure(ResourceScope.Machine)]
+        [ResourceConsumption(ResourceScope.Machine)]
+        static internal Stream GetXmlStreamFromValues(String[] values, String errorString) {
+            if (values.Length != 1){
+                throw ADP.ConfigWrongNumberOfValues(errorString);
+            }
+            return ADP.GetXmlStream(values[0],errorString);
+        }
+
+        // SxS (VSDD 545786): metadata files are opened from <.NetRuntimeFolder>\CONFIG\<metadatafilename.xml>
+        // this operation is safe in SxS because the file is opened in read-only mode and each NDP runtime accesses its own copy of the metadata
+        // under the runtime folder.
+        // This method returns stream to open file, so its ResourceExposure value is ResourceScope.Machine.
+        [ResourceExposure(ResourceScope.Machine)]
+        [ResourceConsumption(ResourceScope.Machine)]
+        static internal Stream GetXmlStream(String value, String errorString) {
+            Stream XmlStream;
+            const string config = "config\\";
+            // get location of config directory
+            string rootPath = System.Runtime.InteropServices.RuntimeEnvironment.GetRuntimeDirectory();
+            if (rootPath == null) {
+                    throw ADP.ConfigUnableToLoadXmlMetaDataFile(errorString);
+            }
+            StringBuilder tempstring = new StringBuilder(rootPath.Length+config.Length+value.Length);
+            tempstring.Append(rootPath);
+            tempstring.Append(config);
+            tempstring.Append(value);
+            String fullPath = tempstring.ToString();
+
+            // don't allow relative paths
+            if (ADP.GetFullPath(fullPath) != fullPath) {
+                throw ADP.ConfigUnableToLoadXmlMetaDataFile(errorString);
+            }
+
+            try {
+                XmlStream = ADP.GetFileStream(fullPath);
+            }
+            catch(Exception e){
+                // UNDONE - should not be catching all exceptions!!!
+                if (!ADP.IsCatchableExceptionType(e)) {
+                    throw;
+                }
+                throw ADP.ConfigUnableToLoadXmlMetaDataFile(errorString);
+            }
+
+            return XmlStream;
+
+        }
+
+        [ResourceExposure(ResourceScope.Machine)]
+        [ResourceConsumption(ResourceScope.Machine)]
+        static internal object ClassesRootRegistryValue(string subkey, string queryvalue) {
+            //(new RegistryPermission(RegistryPermissionAccess.Read, "HKEY_CLASSES_ROOT\\" + subkey)).Assert();
+            try {
+                using(RegistryKey key = Registry.ClassesRoot.OpenSubKey(subkey, false)) {
+                    return ((null != key) ? key.GetValue(queryvalue) : null);
+                }
+            }
+            catch(SecurityException e) {
+                // Even though we assert permission - it's possible there are
```

Comment:
remove "Even though we assert permission - " and the registrypermission assert above. and below - then you can remove the finally.

 You might do some grepping for "Permission" and "Assert" .. if we've already removed "using System.Security.Permissions;" they can only be in comments, though